subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/jar/JarClasspathSnapshotFactory.java

```diff
@@ -39,12 +39,14 @@ JarClasspathSnapshot createSnapshot(Iterable<JarArchive> jarArchives) {
         Set<String> duplicateClasses = Sets.newHashSet();
 
         for (JarArchive jar : jarArchives) {
-            JarSnapshot snapshot = jarSnapshotter.createSnapshot(jar);
-            jarSnapshots.put(jar.file, snapshot);
-            jarHashes.put(jar.file, snapshot.getHash());
-            for (String c : snapshot.getClasses()) {
-                if (!allClasses.add(c)) {
-                    duplicateClasses.add(c);
+            if (jar.file.exists()) {
```

Comment:
Not sure if this is the right place or whether this filtering should be done where this method is called.